src/ArduinoIoTCloudNotecard.cpp

```diff
@@ -0,0 +1,552 @@
+/*
+   This file is part of ArduinoIoTCloud.
+
+   Copyright 2019 ARDUINO SA (http://www.arduino.cc/)
+
+   This software is released under the GNU General Public License version 3,
+   which covers the main part of arduino-cli.
+   The terms of this license can be found at:
+   https://www.gnu.org/licenses/gpl-3.0.en.html
+
+   You can be released from the requirements of the above licenses by purchasing
+   a commercial license. Buying such a license is mandatory if you want to modify or
+   otherwise use the software for commercial activities involving the Arduino
+   software without disclosing the source code of your own applications. To purchase
+   a commercial license, send an email to license@arduino.cc.
+*/
+
+/******************************************************************************
+ * INCLUDE
+ ******************************************************************************/
+
+#include "AIoTC_Config.h"
+
+#if defined(USE_NOTECARD)
+
+#include "ArduinoIoTCloudNotecard.h"
+
+#include <algorithm>
+#include <functional>
+
+#include <Arduino_NotecardConnectionHandler.h>
+
+#include "cbor/CBOREncoder.h"
+#include "cbor/MessageDecoder.h"
+#include "cbor/MessageEncoder.h"
+
+#if OTA_ENABLED
+#include "ota/OTA.h"
+#endif /* OTA_ENABLED */
+
+/******************************************************************************
+ * CONSTANTS
+ ******************************************************************************/
+
+static size_t const CBOR_NOTE_MSG_MAX_SIZE = 255;
+static size_t const POLL_INTERVAL_MS = 10000;
+
+/******************************************************************************
+ * LOCAL MODULE FUNCTIONS
+ ******************************************************************************/
+
+unsigned long getTime()
+{
+  return ArduinoCloud.getInternalTime();
+}
+
+void ISR_dataAvailable(void)
+{
+  ArduinoCloud._data_available = true;
+}
+
+/******************************************************************************
+ * CTOR/DTOR
+ ******************************************************************************/
+
+ArduinoIoTCloudNotecard::ArduinoIoTCloudNotecard()
+:
+  _state{State::ConnectPhy}
+  ,_connection_attempt(0,0)
+  ,_message_stream(std::bind(&ArduinoIoTCloudNotecard::sendMessage, this, std::placeholders::_1))
+  ,_thing(&_message_stream)
+  ,_device(&_message_stream)
+  ,_last_poll_ms{0}
+  ,_interrupt_pin{-1}
+  ,_data_available{false}
+#if defined(BOARD_HAS_SECRET_KEY)
+  ,_secret_device_key{""}
+#endif /* BOARD_HAS_SECRET_KEY */
+#if OTA_ENABLED
+  ,_ota_cap{false}
```

Comment:
This OTA properties are not needed anymore is you are using the command topic

Reply:
Thank you for clarifying. I haven't attempted to implement OTA, but I will be sure to strip these out when I do.